packages/mui-base/src/useAutocomplete/useAutocomplete.js

```diff
@@ -79,6 +79,7 @@ export function useAutocomplete(props) {
   const {
     // eslint-disable-next-line @typescript-eslint/naming-convention
     unstable_isActiveElementInListbox = defaultIsActiveElementInListbox,
+    // TODO should be using base class name by default
```

Comment:
Hooks should not use classes at all.
Besides, we decided yesterday that we'll not use classes as style hooks in Base UI going forward (state will be represented with data attributes).

Reply:
@michaldudak as an outsider (for the time being 😉), I couldn't agree more with this decision. Many states can already be targeted through vanilla CSS pseudoclasses (e.g. `:hover`) and aria attributes (`[aria-disabled="true"]`), so it makes sense that other JS-originated states can be targeted similarly through data attributes. It feels like a natural extension of it.

This is also a useful pattern in the upper layers of a design system stack, even beyond the base headless libraries used under the hood. I did it here, for example: https://github.com/Create-Inc/archetype/blob/b0415b7d7d5be12e67bd3c6bb9ce9c9ae9793558/packages/archetype-ui/src/alert/Alert.tsx#L45-L46

I talked about it on Twitter the other day, in the context of avoiding long `clsx` expressions: https://x.com/daniguardio_la/status/1755684502964175256?s=20

Reply:
> Hooks should not use classes at all.

@michaldudak Agree, in theory the component API should be the one adding the class name. But in this case, I don't see the alternative. I'm not aware we can afford it from a performance perspective.

> Many states can already be targeted through vanilla CSS pseudoclasses (e.g. :hover) and aria attributes ([aria-disabled="true"]),

@DaniGuardiola I think that you are commenting on something different https://github.com/mui/material-ui/pull/26181.

I'm not aware there is plans to use data attributes. As I understand it, they comes with downsides:

-  data attribute value aren't as optimized as class names: https://github.com/mui/material-ui/issues/41068. `&.Base--focused` is going to be more performant than `&. [aria-focused="true"]`
- `&.Base--focused` is easier to write than `&. [data-focused="true"]`

But not really with upsides.

Reply:
@oliviertassinari seems like I misinterpreted it then! I didn't know about the performance implications. The talk linked in that issue is fascinating. Thank you, and sorry for the noise.

Reply:
> I'm not aware there is plans to use data attributes.

@oliviertassinari The Base UI new Base UI API will include data attributes in place of classes: https://www.notion.so/mui-org/base-ui-Style-hooks-e943c3051ca74ce79f2e5af98a0f43eb

Reply:
Ok, I'm removing this comment, keeping the PR more focused on only be about the regression:

```suggestion
```